### PR TITLE
Add exhaustive text splitter

### PR DESCRIPTION
**Pull request by yuriymiquido**

### CODE REVIEW

File: src/draive/splitters/exhaustive.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+from collections.abc import Callable, Sequence
+
+
+def split_text(
```
Comment (KaQuMiQ) [inline src/draive/splitters/exhaustive.py:4]: please rename to i.e. "split_text_exhaustively"

File: src/draive/splitters/exhaustive.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    for part in parts:
+        merged_part: str = accumulator + part
+        # check if can add to previous part
+        if count_size(merged_part) <= part_size:
+            accumulator = merged_part
+            overlap_accumulator.append(part)
+
+        # check if current part is not too big on its own
+        elif count_size(part) > part_size:
+            raise ValueError("Text splitting failed")
```
Comment (KaQuMiQ) [inline src/draive/splitters/exhaustive.py:89]: please change the error description i.e. "Failed to split text  fitting required size."

File: src/draive/splitters/text.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+from collections.abc import Callable, Sequence
+from typing import Literal
+
+from draive.splitters.basic import split_text as basic_split_text
```
Comment (KaQuMiQ) [inline src/draive/splitters/text.py:4]: please rename in the basic file instead of using alias

File: src/draive/splitters/exhaustive.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                    overlap_accumulator = [element, *overlap_accumulator]
+                    accumulator = merged_accumulator
+
+                else:
+                    break
+
+            overlap_accumulator.append(part)
+            accumulator = accumulator + part
+
+        # otherwise make start a new part out of current
```
Comment (KaQuMiQ) [inline src/draive/splitters/exhaustive.py:114]: "# otherwise make start a new part out of current" -> "# otherwise start a new part out of current"

**Comment by yuriymiquido** [comment]
I updated the split method with your suggestion

**Review by KaQuMiQ — APPROVED** [review]

**Pull request merged**